src/libraries/System.Security.Principal.Windows/src/System/Security/Principal/SID.cs

```diff
@@ -452,53 +434,59 @@ private void CreateFromBinaryForm(byte[] binaryForm, int offset)
-nameof(binaryForm));
+                throw new ArgumentException(SR.Format(SR.IdentityReference_InvalidNumberOfSubauthorities, MaxSubAuthorities), nameof(binaryForm));
             }
 
+
             //
             // Make sure the buffer is big enough
             //
 
-            int Length = 1 + 1 + 6 + 4 * binaryForm[offset + 1];
+            int subAuthoritiesLength = binaryForm[offset + 1];
 
-            if (binaryForm.Length - offset < Length)
+            if (subAuthoritiesLength > MaxSubAuthorities)
             {
-                throw new ArgumentException(
-                    SR.ArgumentOutOfRange_ArrayTooSmall,
-nameof(binaryForm));
+                throw new ArgumentOutOfRangeException(
+                    "subAuthoritiesLength",
+                    subAuthoritiesLength,
+                    SR.Format(SR.IdentityReference_InvalidNumberOfSubauthorities, MaxSubAuthorities)
+                );
             }
 
-            Authority =
-                (IdentifierAuthority)(
+            int totalLength = 1 + 1 + 6 + 4 * subAuthoritiesLength;
+
+            if (binaryForm.Length - offset < totalLength)
+            {
+                throw new ArgumentException(SR.ArgumentOutOfRange_ArrayTooSmall, nameof(binaryForm));
+            }
+
+            Span<int> subAuthorities = stackalloc int[MaxSubAuthorities];
+
+            IdentifierAuthority authority = (IdentifierAuthority)(
                 (((long)binaryForm[offset + 2]) << 40) +
                 (((long)binaryForm[offset + 3]) << 32) +
                 (((long)binaryForm[offset + 4]) << 24) +
                 (((long)binaryForm[offset + 5]) << 16) +
                 (((long)binaryForm[offset + 6]) << 8) +
-                (((long)binaryForm[offset + 7])));
-
-            SubAuthorities = new int[binaryForm[offset + 1]];
+                (((long)binaryForm[offset + 7]))
+            );
 
             //
             // Subauthorities are represented in big-endian format
             //
 
-            for (byte i = 0; i < binaryForm[offset + 1]; i++)
+            for (int i = 0; i < subAuthoritiesLength; i++)
             {
-                unchecked
-                {
-                    SubAuthorities[i] =
-                        (int)(
-                        (((uint)binaryForm[offset + 8 + 4 * i + 0]) << 0) +
-                        (((uint)binaryForm[offset + 8 + 4 * i + 1]) << 8) +
-                        (((uint)binaryForm[offset + 8 + 4 * i + 2]) << 16) +
-                        (((uint)binaryForm[offset + 8 + 4 * i + 3]) << 24));
-                }
+                subAuthorities[i] =
+                    (int)(
+                    (((uint)binaryForm[offset + 8 + 4 * i + 0]) << 0) +
+                    (((uint)binaryForm[offset + 8 + 4 * i + 1]) << 8) +
+                    (((uint)binaryForm[offset + 8 + 4 * i + 2]) << 16) +
+                    (((uint)binaryForm[offset + 8 + 4 * i + 3]) << 24)
+                );
```

Comment:
Typically the chop style on this would be what it was.  I understand the symmetry here with the bracing rules, so I'm more pointing it out than suggesting a change.